Source/SwiftLintFramework/Extensions/Structure+SwiftLint.swift

```diff
@@ -17,17 +17,21 @@ extension Structure {
     /// - Parameter byteOffset: Int
     // swiftlint:disable:next valid_docs
     internal func kinds(forByteOffset byteOffset: Int) -> [(kind: String, byteRange: NSRange)] {
-        var results = [(kind: String, byteRange: NSRange)]()
+        return kinds().filter {
```

Comment:
I expect this to be slower than the previous implementation because now we keep navigating in a substructure even if the top structure doesn't contain the range. Just a thing to keep in mind (i.e. we need to run oss-check locally to get reliable results about this one)